servers/quarkus-server/src/test/java/org/projectnessie/server/TestAuthorizationRules.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import io.quarkus.test.security.TestSecurity;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.projectnessie.api.ContentsApi;
+import org.projectnessie.api.TreeApi;
+import org.projectnessie.api.params.CommitLogParams;
+import org.projectnessie.api.params.EntriesParams;
+import org.projectnessie.client.NessieClient;
+import org.projectnessie.client.rest.NessieForbiddenException;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.ContentsKey;
+import org.projectnessie.model.EntriesResponse.Entry;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.model.ImmutableDelete;
+import org.projectnessie.model.ImmutableOperations;
+import org.projectnessie.model.ImmutablePut;
+import org.projectnessie.model.Operations;
+import org.projectnessie.server.authz.NessieAuthorizationTestProfile;
+
+@QuarkusTest
+@TestProfile(value = NessieAuthorizationTestProfile.class)
+class TestAuthorizationRules {
```

Comment:
I think, this test can perfectly live in `nessie-services`, with a mocked `VersionStore` store implementation, so we do not have the "only" test for rule only in `nessie-quarkus`.

Reply:
I think it needs to live in `nessie-quarkus`, otherwise we'll have to pull in quarkus-related dependencies into `nessie-services` (for `QuarkusTest / `TestProfile` / `TestSecurity`)

Reply:
Does this mean we are explicitly saying the reference impl of authz only works in Quarkus? Is that documented anywhere?

Reply:
it's not explicitly documented, but rather implicit since all of the authorization rules are defined in a quarkus config.

Reply:
It should go in the public facing documentation (eg in `site`) for authz. Will the public documetnation be a follow up PR?

Reply:
yep, I was planning to add some docs once the PR is finished/merged